### PR TITLE
refac: peek and dequeue outside of actormessagequeue

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR will remove the coupling between `ActorMessageQueue` and `Bundles`. The overall intent is to increase performance when a `Peek` or a `Dequeue` is performed.
- Retrieving the oldest bundle for a actor is now moved into `BundleRepository`
- Retrieving a bundle is now move into `BundleRepository`
- `ActorMessageQueueTests` have been removed, all noteworthy tests are covered in `WhenEnqueueinOutgoingMessageTests` and `WhenPeekIsRequstedTests`

## References
https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/291
## Checklist
- [x] Should the change be behind a feature flag?
- [x] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [ ] Subsystem test executed (dev_002/dev_003)
- [x] Is there time to monitor state of the release to Production?
- [x] Reference to the task

### CODE REVIEW

File: source/IntegrationTests/Behaviours/IntegrationEvents/GivenCalculationCompletedV1ReceivedForBalanceFixingTests.cs [hunk 1]
```diff
@@ -293,7 +293,7 @@ await _fixture.DatabricksSchemaManager.InsertAsync(
         peekResultsForEnergySupplier.Should().HaveCount(expectedNumberOfPeekResults, "Fee result contains a single gap, which should result in two messages");
 
         // Assert first fee is correct and within expected period
-        var assertForFirstBundle = new AssertNotifyAggregatedMeasureDataJsonDocument(peekResultsForEnergySupplier[0].Bundle);
+        var assertForFirstBundle = new AssertNotifyAggregatedMeasureDataJsonDocument(peekResultsForEnergySupplier[1].Bundle);
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Behaviours/IntegrationEvents/GivenCalculationCompletedV1ReceivedForBalanceFixingTests.cs:296]: 1 != first?   ( assertForFIRSTBundle )

File: source/IntegrationTests/Behaviours/IntegrationEvents/GivenCalculationCompletedV1ReceivedForWholesaleFixingTests.cs [hunk 2]
```diff
@@ -527,36 +527,36 @@ await _fixture.DatabricksSchemaManager.InsertAsync(
...
         assertForThirdBundle.HasPeriod(
             new Period(
                 Instant.FromUtc(2023, 2, 5, 23, 0, 0),
                 Instant.FromUtc(2023, 2, 6, 23, 0, 0)));
         assertForThirdBundle.HasPoints(
+        [
+            new WholesaleServicesPoint(1, 1, 12.757m, 12.757m, CalculatedQuantityQuality.Calculated),
+        ]);
+
+        // Assert third fee is correct and within expected period
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Behaviours/IntegrationEvents/GivenCalculationCompletedV1ReceivedForWholesaleFixingTests.cs:551]:
```suggestion
        // Assert First? fee is correct and within expected period
```

File: source/OutgoingMessages.Domain/Models/Bundles/Bundle.cs [hunk 3]
```diff
@@ -93,9 +93,15 @@ public void Add(OutgoingMessage outgoingMessage)
         CloseBundleIfFull(outgoingMessage.CreatedAt);
     }
 
-    internal void Dequeue()
+    public bool TryDequeue()
     {
-        DequeuedAt = SystemClock.Instance.GetCurrentInstant();
+        if (ClosedAt is not null && PeekedAt is not null)
```
Comment (RasmusGraabaek) [inline source/OutgoingMessages.Domain/Models/Bundles/Bundle.cs:98]: It should not be possible for a bundle to be peeked and not closed?
Hence if PeekedAt != null, then closedAt != null?

File: source/OutgoingMessages.Application/UseCases/PeekMessage.cs [hunk 4]
```diff
@@ -76,13 +79,19 @@ public PeekMessage(
             return null;
         }
 
-        var peekResult = request.DocumentFormat == DocumentFormat.Ebix ? actorMessageQueue.Peek() : actorMessageQueue.Peek(request.MessageCategory);
+        await PeekAndCommitToEnsureBundleIsClosedAsync(request, actorMessageQueue.Id).ConfigureAwait(false);
+
+        var bundle = await _bundleRepository.GetOldestBundleAsync(actorMessageQueue.Id, request.MessageCategory).ConfigureAwait(false); // new context required?
```
Comment (RasmusGraabaek) [inline source/OutgoingMessages.Application/UseCases/PeekMessage.cs:84]: I would assume that a new context is NOT required

File: source/OutgoingMessages.Application/UseCases/PeekMessage.cs [hunk 5]
```diff
@@ -113,14 +122,15 @@ public PeekMessage(
...
         // Right after we call Peek(), we close the bundle. This is to ensure that the bundle wont be added more messages, after we have peeked.
         // And before we are able to update the bundle to closed in the database.
-        var actorMessageQueue = await
-            _actorMessageQueueRepository.ActorMessageQueueForAsync(request.ActorNumber, request.ActorRole).ConfigureAwait(false);
-        var peekResult = actorMessageQueue?.Peek(request.MessageCategory);
-        if (peekResult != null)
+        var bundle = await _bundleRepository.GetOldestBundleAsync(actorMessageQueueId, request.MessageCategory).ConfigureAwait(false);
+        if (bundle != null)
+        {
+            bundle.PeekBundle();
```
Comment (RasmusGraabaek) [inline source/OutgoingMessages.Application/UseCases/PeekMessage.cs:132]:
```suggestion
            bundle.Close();
```

**Review by MadsDue — APPROVED** [review]

**Review by MadsDue — APPROVED** [review]

**Pull request merged**